kong/db/migrations/state.lua

```diff
@@ -145,7 +145,44 @@ end
 local State = {}
 State.__index = State
 
+-- @return nil (no executed migrations for subsystem found) or an array with at least one element like:
+-- { name = "000_base",
+--   cassandra = { up = string, down = function | string | nil },
+--   postgres = { up = string, down = function | string | nil }
+-- },
+local function get_executed_migrations_for_subsystem(self, subsystem_name)
+  if not self.executed_migrations then
+    return nil
+  end
+
+  for _, subsys in ipairs(self.executed_migrations) do
+    if subsys.subsystem == subsystem_name then
+      return subsys.migrations
+    end
+  end
+end
 
+-- @return a table with the following structure:
+-- {
+--   executed_migrations = Subsystem[] | nil
+--   pending_migrations  = Subsystem[] | nil
+--   missing_migrations  = Subsystem[] | nil
+--   new_migrations      = Subsystem[] | nil,
+--   legacy_is_014 = boolean,
+--   needs_bootstrap = boolean,
+-- }
+--
+-- Where Subsystem[] is an array with at least one element like:
+--
+-- { subsystem = "core", -- or some other plugin name, like "acl"
+--   namespace = "kong.db.migrations.core", --  or some other plugin namespace, like "kong.plugins.acl.migrations
+--   migrations = { -- an array with at least one element like:
+--     { name = "000_base",
+--       cassandra = { up = string, down = function | string | nil },
+--       postgres = { up = string, down = function | string | nil }
```

Comment:
`teardown`

Reply:
also, `function | nil`, not `function | string | nil`.